lessons/cryptography/Encoding/encoding.md

```diff
@@ -0,0 +1,118 @@
+
+### Base64 Encoding Example
+As an example, we will encode the following binary representation:  
+
+**01001000 01100101 01101100 01101100 01101111**
+
+First, group the binary data into groups of three 8-bit bytes. It is seen that we can make 1 group (shown in paranthesis) of three 8-bit bytes, and then have some binary data leftover. 
+
+**(01001000 01100101 01101100) 01101100 01101111**
+
+For each complete group, re-group the binary data into four 6-bit characters. For the leftover data, form groups of 6-bit characters, and fill in any remaining spots with 0's. In this example, the last group would have been 1111, so we add two 0's. 
+
+**(010010 000110 010101 101100) (011011 000110 111100)**
+
+Now, using the base64 character set chart, we can look up the characters that are represented by each of these 6-bit characters. The spacing is removed, giving us the following text. 
+
+**SGVs bG8**
+
+Since the last group only had three 6-bit groups, we add 1 **=** as padding to make it a complete set of four 6-bit base64 characters, giving a final base64 representation shown below. 
+
+**SGVsbG8=**
+
+### Base64 Encoding and Decoding with CyberChef
+Although base64 is commonly used for encoding binary data into ASCII text, it is also used to encode other character based messages into ASCII text as well. For example, if you want to base64 encode the message **"base64 encoding!"**, that message would first be encoded into binary using ASCII encoding, and then base64 encoded using the process previously described. The image below shows the message base64 encoded using CyberChef. 
+
+![Base64 Encoding with CyberChef (https://gchq.github.io/CyberChef/#recipe=To_Base64('A-Za-z0-9%2B/%3D')&input=YmFzZTY0IGVuY29kaW5nIQ)](.rsrc/cyberchef-base64-encode.png) 
+
+The image below then shows the base64 encoded message from the previous step getting decoded using CyberChef. 
+
+![Base64 Encoding with CyberChef (https://gchq.github.io/CyberChef/#recipe=From_Base64('A-Za-z0-9%2B/%3D',true,false)&input=WW1GelpUWTBJR1Z1WTI5a2FXNW5JUT09)](.rsrc/cyberchef-base64-decode.png) 
+
+
+# The Real World, Prolific Breaches
+Base64 encoding and decoding is commonly used in web applications. For example, an image can be encoded with base64, and then embedded in HTML. Looking at the sample code snippet below, you can see a string of characters that looks like it is encoded with base64. Notice that the string ends with **==**, which is often a characteristic of base64 encoded characters.  
+
+![Embedding a Base64 Encoded Image into HTML (https://www.w3docs.com/snippets/html/how-to-display-base64-images-in-html.html)](.rsrc/base64-html.png) 
+
+The above code snippet uses the **img** tag in HTML to embed the picture that is encoded with base64. 
+
+# Check YoSelf
+## Q1 - Use base64 to encode the word **cyber** using CyberChef (or manually for an extra challenge). 
```

Comment:
great, just need some wrong answers here as well, four total options for each multiple choice, please.

Reply:
maybe binary encode one, morse encode another, base64 a wrong pt for a third?

Reply:
done